File: 04.Simulation模拟/leetcode_1886_findRotation.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
private:
    int n;
    vector<vector<int>> rotate(vector<vector<int>>& mat){
        vector<vector<int>> newMat(n, vector<int>(n, 0));
        for (int i=0; i<n; i++){
            for (int j=0; j<n; j++){
                newMat[i][j] = mat[n-j-1][i];
            }
        }
        return newMat;
    }

public:
    bool findRotation(vector<vector<int>>& mat, vector<vector<int>>& target) {
        n = mat.size();
        for (int i=0; i<4; i++){
            if (mat == target){
                return true;
            }
            mat = rotate(mat);
        }
        return false;
    }
};

int main(){
    Solution s;
    vector<vector<int>> mat = {{0,1},{1,0}};
    vector<vector<int>> target = {{1,0},{0,1}};
    cout << s.findRotation(mat, target) << endl;
}

File: 04.Simulation模拟/leetcode_1886_findRotation.py
from typing import List

class Solution:
    def findRotation(self, mat: List[List[int]], target: List[List[int]]) -> bool:
        n = len(mat)
        def rotate(mat):
            newMat = [[0]*n for _ in range(n)]
            for i in range(n):
                for j in range(n):
                    newMat[i][j] = mat[n-j-1][i]
            return newMat
        
        for _ in range(4):
            if mat == target:
                return True
            mat = rotate(mat)
        return False

if __name__ == "__main__":
    s = Solution()
    mat = [[0,1],[1,0]]
    target = [[1,0],[0,1]]
    print(s.findRotation(mat, target))
